test/geometry/transform/test_pyramid.py

```diff
@@ -192,3 +192,31 @@ def test_gradcheck(self, device, dtype):
             raise_exception=True,
             fast_mode=True,
         )
+
+
+class TestUpscaleDouble:
+    @pytest.mark.parametrize("shape", ((10, 10), (3, 10, 10), (1, 3, 10, 10)))
+    def test_upscale_double(self, shape, device, dtype):
+        print(f"shape: {shape}")
+        print(f"device: {device}")
+        print(f"dtype: {dtype}")
+
+        x = torch.zeros(shape, device=device, dtype=dtype)
+
+        xa = torch.arange(0, 10, step=1, dtype=int)
+        ya = torch.arange(0, 10, step=1, dtype=int)
+        xm, ym = torch.meshgrid(xa, ya, indexing='ij')
+
+        if len(shape) == 2:
+            x = xm
+        else:
+            x[..., 0, :, :] = xm
+            x[..., 1, :, :] = ym
+
+        upscaled = kornia.geometry.transform.upscale_double(x)
+        assert torch.all(
+            upscaled[..., 1::2, 1::2][..., :-1, :-1]
+            == (upscaled[..., ::2, ::2][..., :-1, :-1] + upscaled[..., ::2, ::2][..., 1:, 1:]) / 2
+        )
+        downscaled_back = upscaled[..., ::2, ::2]
+        assert torch.all(x == downscaled_back)
```

Comment:
```suggestion
        assert_close(x, downscaled_back)
```

`from kornia.testing import assert_close`